### PR TITLE
Get the list of teams in an organization.

### PR DESCRIPTION
**Pull request by anamariamv**
This PR solves #337

### CODE REVIEW

File: github4s/src/main/scala/github4s/domain/Team.scala [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package github4s.domain
+
+case class Team(
```
Comment (juanpedromoreno) [inline github4s/src/main/scala/github4s/domain/Team.scala]:
```suggestion
final case class Team(
```

File: github4s/src/main/scala/github4s/interpreters/TeamsInterpreter.scala [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+package github4s.interpreters
+
+import cats.Applicative
+import github4s.Decoders._
+import github4s.GithubResponses.GHResponse
+import github4s.algebras.Teams
+import github4s.domain._
+import github4s.http.HttpClient
+
+class TeamsInterpreter[F[_]: Applicative](
```
Comment (BenFradet) [inline github4s/src/main/scala/github4s/interpreters/TeamsInterpreter.scala]: is the `Applicative` constraint needed?

File: github4s/src/test/scala/github4s/utils/BaseIntegrationSpec.scala [hunk 3]
```diff
@@ -27,6 +27,7 @@ import github4s.integration.{
   GHOrganizationsSpec,
   GHPullRequestsSpec,
   GHReposSpec,
+  GHTeamsSpec,
```
Comment (BenFradet) [inline github4s/src/test/scala/github4s/utils/BaseIntegrationSpec.scala]: :art: maybe we can import all here

Comment (anamariamv) [reply github4s/src/test/scala/github4s/utils/BaseIntegrationSpec.scala]: 👍

File: github4s/src/main/scala/github4s/algebras/Teams.scala [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package github4s.algebras
+
+import github4s.GithubResponses.GHResponse
+import github4s.domain.{Pagination, Team}
```
Comment (bond15) [inline github4s/src/main/scala/github4s/algebras/Teams.scala]: opinion: the rest of the algebras import github4s.domain._

File: github4s/src/main/scala/github4s/Decoders.scala [hunk 5]
```diff
@@ -278,4 +278,37 @@ object Decoders {
           starred_at <- c.downField("starred_at").as[String]
           user       <- c.downField("user").as[User]
         } yield Stargazer(Some(starred_at), user)))
+
+  implicit val decodeTeam: Decoder[Team] =
```
Comment (bond15) [inline github4s/src/main/scala/github4s/Decoders.scala]: Can the team decoder be automatically derived?

implicit val decoderTeam: Decoder[Team]               = deriveDecoder[Team]

**Pull request merged**